evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -868,6 +960,85 @@ where
         }
     }
 
+    /// Create two-to-one block aggregation circuit.
+    ///
+    /// # Arguments
+    ///
+    /// - `block_circuit`: circuit data for the block circuit, that constitues
+    ///   the base case for aggregation.
+    ///
+    /// # Outputs
+    ///
+    /// Returns a [`TwoToOneBlockCircuitData<F, C, D>`].
+    fn create_two_to_one_block_circuit(
+        block_circuit: &BlockCircuitData<F, C, D>,
+    ) -> TwoToOneBlockCircuitData<F, C, D>
+    where
+        F: RichField + Extendable<D>,
+        C: GenericConfig<D, F = F>,
+        C::Hasher: AlgebraicHasher<F>,
+    {
+        let mut builder = CircuitBuilder::<F, D>::new(block_circuit.circuit.common.config.clone());
+
+        let mix_hash = builder.add_virtual_hash_public_input();
+
+        // We need to pad by PIS to match the count of PIS of the `base_proof`.
+        let mut padding = block_circuit.circuit.common.num_public_inputs;
+        // The number of PIS that will be added *after* padding by
+        // [`add_verifier_data_public_inputs()`].
+        padding -= verification_key_len(&block_circuit.circuit);
+        // Account for `mix_pv_hash`.
+        padding -= builder.num_public_inputs();
+
+        for _ in 0..padding {
+            let target = builder.zero();
+            builder.register_public_input(target);
+        }
+
+        let user_pis_len = builder.num_public_inputs();
+        let cyclic_vk = builder.add_verifier_data_public_inputs();
+
+        let lhs = Self::add_agg_child(&mut builder, &block_circuit.circuit);
+        let rhs = Self::add_agg_child(&mut builder, &block_circuit.circuit);
+
+        let lhs_public_values = lhs.public_inputs(&mut builder);
```

Comment:
For clarity: what you get is the entire public inputs, which is slightly longer than the public values.

Reply:
Now I see.  I have proposed a `PublicValuesTarget::SIZE` field.

Reply:
> Now I see. I have proposed a PublicValuesTarget::SIZE field.

Thanks for this, I was actually surprised we didn't have any yet for this struct when doing the review.